src/EditorFeatures/Core/Implementation/AutomaticCompletion/AbstractAutomaticLineEnderCommandHandler.cs

```diff
@@ -51,6 +51,11 @@ internal abstract class AbstractAutomaticLineEnderCommandHandler :
         /// </summary>
         protected abstract void FormatAndApply(Document document, int position, CancellationToken cancellationToken);
 
+        /// <summary>
+        /// special cases where we do not want to do line completion but just fall back to line break and formatting.
+        /// </summary>
+        protected abstract bool TreatAsReturn(Document document, int position, CancellationToken cancellationToken);
```

Comment:
Super duper nit: ```TreatAsBreakLine``` (it's not guaranteed to be Return, right?). Or, the negation, ```ShouldCompleteLine```?